### ISSUE TITLE
Please do not stupidly overwrite MSYS2 if it already exists.

### ISSUE THREAD

**Issue by BenKey** [chocolatey/choco#2091]
Do not stupidly overwrite MSYS2 if it already exists.

I had MSYS2 installed with all the latest packages. I ran "choco upgrade all -y". One of the packages that were upgraded depended on MSYS2. Choco upgrade stupidly ran msys2-x86_64-20161025.exe which installed ancient outdated packages over my already up to date MSYS2 installation.

To make matters worse I cannot fix the stupid mistake made by "choco upgrade all -y" because the "pacman -Syuu" command now fails with numerous errors.

Because of this stupidity, I am going to have to uninstall MSYS2 and reinstall it and all my packages.

If a package depends on MSYS2 either install it to an alternative location or do not overwrite if the MSYS2 directory already exists.

**Comment by pauby** [comment]
> One of the packages that were upgraded depended on MSYS2

Which package was that?

**Comment by BenKey** [comment]
I think I have figured out what happened. Unfortunately I do not know precisely which package cause the old MSYS to be installed since I did not keep the log file or the console window output so I cannot be certain.

I think older versions of Chicken Scheme had msys2-x86_64-20161025 as a dependency. When I ran 'choco upgrade all -y' Chicken Scheme was updated.

If this is in fact what happened this issue has already been resolved. The Chicken Scheme package currently depends on msys2 which installs to an alternate location in order to avoid conflicts with MSYS2 installed outside of the context of chocolatey,

Thus this issue is most likely closed.

I apologize for the tone of my original bug report.
